src/io/procops.c

```diff
@@ -1094,22 +1094,21 @@ MVMint64 MVM_proc_getppid(MVMThreadContext *tc) {
 
 /* generates a random int64 */
 MVMint64 MVM_proc_rand_i(MVMThreadContext *tc) {
-    MVMuint64 result = tinymt64_generate_uint64(tc->rand_state);
-    return *(MVMint64 *)&result;
+    return (MVMint64)jfs64_generate_uint64(tc->rand_state);
 }
 
 /* generates a number between 0 and 1 */
 MVMnum64 MVM_proc_rand_n(MVMThreadContext *tc) {
-    return tinymt64_generate_double(tc->rand_state);
+    return ((jfs64_generate_uint64(tc->rand_state) >> 11) / (double)(1L << 53));
 }
 
 MVMnum64 MVM_proc_randscale_n(MVMThreadContext *tc, MVMnum64 scale) {
-    return tinymt64_generate_double(tc->rand_state) * scale;
+    return ((jfs64_generate_uint64(tc->rand_state) >> 11) / (double)(1L << 53)) * scale;
```

Comment:
Is there a way to get the scaled number without losing so much accuracy? Feels like a shame to throw away those random bits.

Reply:
No idea. Maybe something like shifting by ‘11 - magnitude(scale)’ and then dividing by ‘max_representable_int * scale’ would work, but I don’t know the math well enough to be sure that’s safe and correct.

Reply:
Based on https://stackoverflow.com/questions/46901022/how-to-convert-a-uint64-t-to-a-double-float-between-0-and-1-with-maximum-accurac (and some other things I've read) and our discussion, I think there isn't actually any more precision to be gained.